PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:3 + add:0, change matches:
-    _inherit = ['pos.order']
-    _inherit = ['pos.payment']
-    _inherit = ['pos.payment.method']

Total Changes: 6

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/pos_restaurant_adyen/models/pos_order.py
+++ b/addons/pos_restaurant_adyen/models/pos_order.py
@@ -5,7 +5,7 @@ from odoo import models
 
 
 class PosOrder(models.Model):
-    _inherit = ['pos.order']
+    _inherit = 'pos.order'
 
     def action_pos_order_paid(self):
         res = super(PosOrder, self).action_pos_order_paid()

--- a/addons/pos_restaurant_adyen/models/pos_payment.py
+++ b/addons/pos_restaurant_adyen/models/pos_payment.py
@@ -10,7 +10,7 @@ TIMEOUT = 10
 
 
 class PosPayment(models.Model):
-    _inherit = ['pos.payment']
+    _inherit = 'pos.payment'
 
     def _update_payment_line_for_tip(self, tip_amount):
         """Capture the payment when a tip is set."""

--- a/addons/pos_restaurant_adyen/models/pos_payment_method.py
+++ b/addons/pos_restaurant_adyen/models/pos_payment_method.py
@@ -5,7 +5,7 @@ from odoo import fields, models, api
 
 
 class PosPaymentMethod(models.Model):
-    _inherit = ['pos.payment.method']
+    _inherit = 'pos.payment.method'
 
     adyen_merchant_account = fields.Char(help='The POS merchant account code used in Adyen')
 
